x-pack/plugins/security_solution/server/usage/types.ts

```diff
@@ -33,6 +32,35 @@ export interface AlertAggs {
   };
 }
 
-export interface RuleSearchResult {
-  alert: SanitizedAlert<RuleParams>;
-}
+/**
+ * This type is _very_ similar to "RawRule". However, that type is not exposed in a non-restricted-path
+ * and it does not support generics well. Trying to use "RawRule" directly with TypeScript Omit does not work well.
+ * If at some point the rules client API supports cross spaces for gathering metrics, then we should remove our use
+ * of SavedObject types and this type below and instead natively use the rules client.
+ *
+ * NOTE: There are additional types not expressed below such as "apiKey" or there could be other slight differences
+ * but this will the easiest way to keep these in sync and I see other code that is similar to this pattern.
+ * {@see RawRule}
+ */
+export type RuleSearchResult = Omit<
+  SanitizedAlert<RuleParams>,
+  'createdBy' | 'updatedBy' | 'createdAt' | 'updatedAt'
+> & {
+  createdBy: string | null;
+  updatedBy: string | null;
+  createdAt: string;
+  updatedAt: string;
+};
+
+/**
+ * Same as "RuleSearchResult" just a partial is applied. Useful for unit tests since these types aren't exact.
+ */
+export type PartialRuleSearchResult = Omit<
```

Comment:
Dead type. Already removed in a follow up [commit](https://github.com/elastic/kibana/pull/124912/commits/5778b9764e8d7bd021e972e1da9897155a047812)